VBA/Office-Shared-VBA/articles/about-autosave.md

```diff
@@ -0,0 +1,120 @@
+
+#### Workarounds
+
+- Consider removing code that writes to the workbook in **BeforeSave** or **AfterSave** events. For example, the add-in described above might be modified to store the change log in a separate file or database.
+
+### Issue 4: Code that dirties the workbook in **AfterSave** (Excel Only)
+
+#### Description
+
+When AutoSave is on, the **BeforeSave** and **AfterSave** events will only trigger if there has been a change in the workbook since the last time they were triggered. If code in the **AfterSave** event dirties the workbook (i.e. makes additional changes), that could potentially trigger events to fire again after a while for the same change then queue up the events to fire again, and so on, indefinitely. This could waste system resources and battery life.
+
+#### Workarounds
+
+- Code that dirties the workbook in **AfterSave** should be moved to **BeforeSave** or to another location entirely (see Issue 3 above). This isn't good practice today, even without AutoSave, since it leaves the workbook in a perpetual "dirty" state, which causes a prompt to appear on close that asks the user to save their changes even if they made no additional changes. 
+
+### Issue 5: Code that cancels the file save in **BeforeSave** (by setting Cancel argument to True)
+
+#### Description
+
+Today, it is possible to cancel the save in the **BeforeSave** event by setting cancel to True: 
+```vb
+Private Sub Workbook_BeforeSave(ByVal SaveAsUI As Boolean, Cancel As Boolean) 
+    Cancel = True 
+End Sub
+```
+When AutoSave is enabled, the application (that is, Excel, Word, or PowerPoint) will trigger saves automatically on a continuous basis until the file has no more unsaved changes. After a single change is made to the file by the user, the application will attempt to save it. If the developer chooses to cancel the save in the manner described above, the application continually determines that there is are unsaved changes, which will cause the save to (eventually) be attempted again. Since the same event code that cancelled the first save will also cancel this second save attempt, the process will continue for as long as the file is open, potentially degrading performance and battery life.
+
+#### Example Scenario
+
+An add-in might completely override the default Word save code so that the file is actually saved to a corporate database instead of a disk or sharepoint location. Such an add-in would first cancel the attempted save before trying to save in another place.
+
+#### Workarounds
+
+- Such add-ins should ensure that AutoSave is turned off by setting autoSaveOn to False. Since a file must already be saved in a OneDrive or SharePoint location for AutoSave to be on, AutoSave should already be off in most cases in the scenario above.
+
+## Example
```

Comment:
This example seems unrelated to the above content. Should it be moved to the beginning to be with the general sentence about "turning off autosave" being a general workaround to all these problems.